hydradx/tests/test_hollar.py

```diff
@@ -7,353 +8,510 @@
+        pool.swap(agent, tkn_buy=tkn_sell, tkn_sell=tkn, buy_quantity=sell_quantity)
+        if hsm.fail or pool.fail:
+            raise ValueError("Swap failed")
+        max_buy_amt2, buy_price2 = hsm.get_buy_params(tkn)
+        remaining_capacity = max_buy_amt - sell_quantity
+        if max_buy_amt2 != pytest.approx(remaining_capacity, rel=1e-20):
+            raise ValueError("Max buy amount changed")
+        if buy_price2 != buy_price:
+            raise ValueError("Buy price changed")
+        # check that update resets block
+        hsm.update()
+        if hsm.fail:
+            raise ValueError("Update failed")
+        max_buy_amt3, buy_price3 = hsm.get_buy_params(tkn)
+        if max_buy_amt3 >= max_buy_amt:
+            raise ValueError("Max buy amount should be less than before swap")
+        if max_buy_amt3 <= max_buy_amt2:
+            raise ValueError("Max buy amount should be greater than after swap but before update")
+        if buy_price3 <= buy_price:
+            raise ValueError("Buy price should be greater than before swap")
+
+
+def test_peg_updated_during_block():
+    peg_raise_n = 10
+    daily_return = 0.01  # 1% return in a day
+    daily_blocks = 7200
+    mult_inc = math.pow(1 + daily_return, peg_raise_n / daily_blocks)
+    ratios = [mpf(0.5), mpf(0.5)]
+    susds_price = 1.5
+    buyback_speed = mpf(1/10_000)
+    liquidity = {'USDT': mpf(1_000_000), 'SUSDS': mpf(1_000_000)}
+    usdt_pool = StableSwapPoolState({'USDT': ratios[0] * mpf(1_000_000), 'HOLLAR': mpf(1_000_000)},100, trade_fee=0.0001,)
+    susds_pool = StableSwapPoolState({'HOLLAR': 1_000_000, 'SUSDS': ratios[1] * 1_000_000 / susds_price},100,
+                                     trade_fee=0.0001, peg=susds_price)
+    pools = [usdt_pool, susds_pool]
+    sell_fee = 0.001
+    max_buy_price_coef = 0.999
+    buy_fee = 0.0001
+    hsm = StabilityModule(liquidity, buyback_speed, pools, sell_fee, max_buy_price_coef, buy_fee)
+    hsm2 = copy.deepcopy(hsm)
+    profit_diff = []
+    for t in range(daily_blocks):
+        if t % peg_raise_n == 0:
+            # do arb in first pool
+            agent = Agent()
+            hsm.arb(agent, 'SUSDS')
+            # update peg target for both pools
+            old_peg_target = hsm.pools['SUSDS'].peg_target[1]
+            assert old_peg_target == hsm2.pools['SUSDS'].peg_target[1]
+            hsm.pools['SUSDS'].set_peg_target(old_peg_target * mult_inc)
+            hsm2.pools['SUSDS'].set_peg_target(old_peg_target * mult_inc)
+            hsm.update()  # pushes update to internal pool_state object
+            hsm2.update()
+            # do arb in second pool
+            agent2 = Agent()
+            hsm2.arb(agent2, 'SUSDS')
+            # calculate profit difference
+            profit_diff.append(agent2.get_holdings('SUSDS') - agent.get_holdings('SUSDS'))
+    total_profit_diff = sum(profit_diff)
+    print('done')
```

Comment:
Shouldn't this assert something?

Reply:
There is an assert, it is checking that the peg_target doesn't change due to the hsm.arb() function being called. The assert is in the for loop.